src/java.base/unix/native/libnio/ch/FileChannelImpl.c

```diff
@@ -176,7 +176,27 @@ Java_sun_nio_ch_FileChannelImpl_transferTo0(JNIEnv *env, jobject this,
 
 #if defined(__linux__)
     off64_t offset = (off64_t)position;
-    jlong n = sendfile64(dstFD, srcFD, &offset, (size_t)count);
+
+    jlong n;
+    if (my_copy_file_range_func != NULL) {
+        n = my_copy_file_range_func(srcFD, &offset, dstFD, NULL, count, 0);
```

Comment:
I assume count will need to be cast to size_t for this to compile on all architectures.

Reply:
Fixed in ee29211e7ebdfc6d9127bcf093de2a475f9c1a3a.